### PR TITLE
feat: add MariaDB extractor

### PR DESCRIPTION
**Pull request by scortier**
## Deliverables

- [x] add unit tests
- [x] add extractor
- [x] add README.md in ```plugins/extractors/mariadb```, defining output
- [x] register your extractor ```plugins/extractors/populate.go```
- [x] add extractor the extractor list in ```docs/reference/extractor.md```

### Table
| Field | Sample Value |
| :---- | :---- |
| `urn` | `my_database.my_table` |
| `name` | `my_table` |
| `source` | `mariadb` |
| `description` | `table description` |
| `profile.total_rows` | `2100` |
| `schema` | [][Column] |

### Column
| Field | Sample Value |
| :---- | :---- |
| `name` | `total_price` |
| `description` | `item's total price` |
| `data_type` | `decimal` |
| `is_nullable` | `true` |
| `length` | `12,2` |

#### Issue #233

### CODE REVIEW

File: plugins/extractors/mariadb/mariadb.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
+package mariadb
+
+import (
+	"context"
+	"database/sql"
+	_ "embed" // used to print the embedded assets
+	"fmt"
+	_ "github.com/go-sql-driver/mysql" // used to register the mariadb driver
```
Comment (GrayFlash) [inline plugins/extractors/mariadb/mariadb.go:8]: fix the driver here

Comment (scortier) [reply plugins/extractors/mariadb/mariadb.go:8]: Mariadb supports the same driver as of mysql

File: plugins/extractors/mariadb/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# mariadb
+
+## Usage
+
+```yaml
+source:
+  type: mariadb
+  config:
+    user_id: admin
```
Comment (mabdh) [inline plugins/extractors/mariadb/README.md:9]: @scortier should we use connection url for this instead?

Comment (scortier) [reply plugins/extractors/mariadb/README.md:9]: sure, will change it

**Review by mabdh — APPROVED** [review]

**Review by ravisuhag — APPROVED** [review]

**Pull request merged**